File: 90. Subsets II/90. Subsets II.dart
class Solution {
  List<List<int>> subsetsWithDup(List<int> nums) {
    nums.sort();
    List<List<int>> result = [];
    backtrack(nums, 0, [], result);
    return result;
  }
  
  void backtrack(List<int> nums, int start, List<int> current, List<List<int>> result) {
    result.add(List.from(current));
    
    for (int i = start; i < nums.length; i++) {
      if (i > start && nums[i] == nums[i-1]) continue;
      current.add(nums[i]);
      backtrack(nums, i + 1, current, result);
      current.removeLast();
    }
  }
}

File: 90. Subsets II/90. Subsets II.cs
public class Solution {
    public IList<IList<int>> SubsetsWithDup(int[] nums) {
        Array.Sort(nums);
        var result = new List<IList<int>>();
        Backtrack(nums, 0, new List<int>(), result);
        return result;
    }
    
    private void Backtrack(int[] nums, int start, List<int> current, List<IList<int>> result) {
        result.Add(new List<int>(current));
        
        for (int i = start; i < nums.Length; i++) {
            if (i > start && nums[i] == nums[i-1]) continue;
            current.Add(nums[i]);
            Backtrack(nums, i + 1, current, result);
            current.RemoveAt(current.Count - 1);
        }
    }
}